api/layer/layer.go

```diff
@@ -323,10 +322,12 @@ func (n *layer) Owner(ctx context.Context) user.ID {
 	return ownerID
 }
 
-func (n *layer) prepareAuthParameters(ctx context.Context, prm *neofs.PrmAuth) {
+func (n *layer) prepareAuthParameters(ctx context.Context, prm *neofs.PrmAuth, bktOwner user.ID) {
 	if bd, ok := ctx.Value(api.BoxData).(*accessbox.Box); ok && bd != nil && bd.Gate != nil {
-		prm.BearerToken = bd.Gate.BearerToken
-		return
+		if issuer, ok := bd.Gate.BearerToken.Issuer(); ok && bktOwner.Equals(issuer) {
```

Comment:
It seems we can adopt https://github.com/nspcc-dev/neofs-s3-gw/pull/485

Reply:
I would like to rebase `tree-service` branch after this PR. Adopt it in separate PR.

Reply:
I messed up with the branches. Both SDK update and this PR were into `master` branch, so we could adopt it. For some reason I thought this PR has `tree-service` target.

I pushed https://github.com/nspcc-dev/neofs-s3-gw/commit/4ed939773b020ee6e305f9123b53e62cdbd6f859 into master branch.